### PR TITLE
✨ Source-Stripe: Add incremental stream support to `accounts` and to `customer_balance_transactions` streams

### PR DESCRIPTION
**Pull request by williamkaper**
## What
This PR adds incremental sync support to the `accounts` stream (I did not create an issue because I figured I'd just submit the change via PR) and to the `customer_balance_transactions` stream. I also updated `transfer_reversals` to be an incremental sub stream.

## How
I followed the pattern for other `UpdatedCursorIncrementalStripeStreams `and changed it from `StripeStream` to `UpdatedCursorIncrementalStripeStream `keying off of the events for a new updated cursor column that I added to the ERD. I also rewrote the customer_balance_transactions stream implementation to go from being something completely bespoke to a substream. Lastly, I updated transfer_reversals to be an incremental sub stream. 

## Review guide

1. erd/discovered_catalog.json - Added new columns and new sync types to accouts
2. erd/source.dbml - Added new columns to erd
3. integration_tests/configured_catalog.json
4. source_strip/schemas/accounts.json - added updated column
6. source.py -- Added new streaming operations for accounts stream
7. docs/stripe.md -- Fixed a bad link that was in main
8. PyProject.toml and metadata.yaml updated

**** NOTE:** I did not update the unit_tests/integration for test_accounts.py. I was going to use one of the other tests in the holder to try to rewrite it, but it was beyond me. The tests are passing, but that test suite isn't testing the incremental part of the stream. I'm hoping a member of the Airbyte Stripe source connector team can pick that up. 

## User Impact
For merchants with large amounts of connected accounts, the stream update won't take hours to keep repulling needlessly.

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: docs/integrations/sources/stripe.md [hunk 1]
```diff
@@ -127,7 +127,7 @@ The Stripe source connector supports the following streams:
 - [Transactions](https://stripe.com/docs/api/transfers/list) \(Incremental\)
 - [Transfers](https://stripe.com/docs/api/transfers/list) \(Incremental\)
 - [Transfer Reversals](https://stripe.com/docs/api/transfer_reversals/list)
-- [Usage Records](https://stripe.com/docs/api/usage_records/subscription_item_summary_list)
```
Comment (williamkaper) [inline docs/integrations/sources/stripe.md:130]: This was causing the tests to fail - not sure how this got here...

**Comment by natikgadzhi** [comment]
Just calling out that I was slow this week, but someone on @airbytehq/dev-python will take a look in the next week, if we have a chance.

**Review by alafanechere — CHANGES_REQUESTED** [review]
Hey @williamkaper, thank you for this awesome contribution!
I'm requesting changes because I noticed a regression while running our internal test suite: when set in `full_refresh` mode the `customer_balance_transaction` stream is not producing any record. I can't identify the root cause of it, I'm quite new to this connector. 
I tried switching the class to `UpdatedCursorIncrementalStripeSubStream` instead of `UpdatedCursorIncrementalStripeLazySubStream` and the missing records get produced.
So it has something to do with the "lazyness".
Let me know if you can reproduce this problem.

@maxi297 is it legit to not use a lazy substream in this situation?

File: docs/integrations/sources/stripe.md [hunk 2]
```diff
@@ -239,6 +239,7 @@ Each record is marked with `is_deleted` flag when the appropriate event happens
 
 | Version | Date       | Pull Request                                              | Subject                                                                                                                                                                                                                       |
 |:--------|:-----------|:----------------------------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 5.6.3 | 2024-10-12 | [46864](https://github.com/airbytehq/airbyte/pull/46864) | Add incremental stream support to `accounts` stream |
```
Comment (alafanechere) [inline docs/integrations/sources/stripe.md:242]:
```suggestion
| 5.7.0 | 2024-10-12 | [46864](https://github.com/airbytehq/airbyte/pull/46864) | Add incremental stream support to `accounts` and `customer_balance_transactions` streams |
```

**Comment by williamkaper** [comment]
> Hey @williamkaper, thank you for this awesome contribution! I'm requesting changes because I noticed a regression while running our internal test suite: when set in `full_refresh` mode the `customer_balance_transaction` stream is not producing any record. I can't identify the root cause of it, I'm quite new to this connector. I tried switching the class to `UpdatedCursorIncrementalStripeSubStream` instead of `UpdatedCursorIncrementalStripeLazySubStream` and the missing records get produced. So it has something to do with the "lazyness". Let me know if you can reproduce this problem.
> 
> @maxi297 is it legit to not use a lazy substream in this situation?

How can I see the test failure easily? When I run airbyte-ci connectors --name source-stripe test I seem to get different transient results, but the actual test failure isn't cleanly surfaced

**Comment by alafanechere** [comment]
> @alafanechere How can I see the test failure easily?

@williamkaper unfortunately you can't ATM. We run a regression test suite on our sandbox accounts. The reports are private because they might expose sensitive information.
Did you try running `customer_balance_transactions` in `full_refresh` in production?

**Comment by williamkaper** [comment]
> > @alafanechere How can I see the test failure easily?
> 
> @williamkaper unfortunately you can't ATM. We run a regression test suite on our sandbox accounts. The reports are private because they might expose sensitive information. Did you try running `customer_balance_transactions` in `full_refresh` in production?

It doesn't appear full_refresh works. Do you want to change the stream type or want me to change the stream type?

**Comment by williamkaper** [comment]
> > > @alafanechere How can I see the test failure easily?
> > 
> > 
> > @williamkaper unfortunately you can't ATM. We run a regression test suite on our sandbox accounts. The reports are private because they might expose sensitive information. Did you try running `customer_balance_transactions` in `full_refresh` in production?
> 
> It doesn't appear full_refresh works. Do you want to change the stream type or want me to change the stream type?

I just pushed it up

**Comment by williamkaper** [comment]
@alafanechere where are we at on this? Is there anything you are waiting for me to do or is this now on your team's conveyor belt?

**Review by alafanechere — APPROVED** [review]
I made some fixes to get CI 🟢 .
Thanks @williamkaper for your contribution and patience!

**Pull request merged**